chef-config/lib/chef-config/fips.rb

```diff
@@ -21,33 +21,11 @@
 module ChefConfig
 
   def self.fips?
-    if ChefUtils.windows?
-      begin
-        require "win32/registry" unless defined?(Win32::Registry)
-      rescue LoadError
-        return false
-      end
-
-      # from http://msdn.microsoft.com/en-us/library/windows/desktop/aa384129(v=vs.85).aspx
-      reg_type =
-        case ::RbConfig::CONFIG["target_cpu"]
-        when "i386"
-          Win32::Registry::KEY_READ | 0x100
-        when "x86_64"
-          Win32::Registry::KEY_READ | 0x200
-        else
-          Win32::Registry::KEY_READ
-        end
-      begin
-        Win32::Registry::HKEY_LOCAL_MACHINE.open('System\CurrentControlSet\Control\Lsa\FIPSAlgorithmPolicy', reg_type) do |policy|
-          policy["Enabled"] != 0
-        end
-      rescue Win32::Registry::Error
-        false
-      end
+    require "openssl" unless defined?(OpenSSL)
+    if defined?(OpenSSL.fips_mode) && OpenSSL.fips_mode
+      true
     else
-      fips_path = "/proc/sys/crypto/fips_enabled"
-      File.exist?(fips_path) && File.read(fips_path).chomp != "0"
+      false
     end
```

Comment:
kill the if-else and just return the boolean expression

Reply:
(there should be a rubocop rule for this if there isn't one already

Reply:
someone needs to write a rule to finding unmatched parens in comments as well)

Reply:
(although someone will ban multiple nesting levels of parents (and i'd hate that))

Reply:
*parens

Reply:
Fixed. I wonder how that ended up that way in Ohai. There must have been more there to start.

Reply:
> there should be a rubocop rule for this if there isn't one already

My continual rubocop experience is being surprised it doesn't handle all of my annoyances.